Travel Scheduler/View Controllers/Place.m

```diff
@@ -15,78 +15,61 @@
 @import GooglePlaces;
 
 typedef void (^getHubDictionaryCompletion)(NSDictionary *, NSError *);
-typedef void (^getNearbyPlacesOfTypeDictionariesCompletion)(NSArray *, NSError *);
-typedef void (^getPhotoOfPlaceCompletion)(NSURL *, NSError *);
+typedef void (^getNearbyPlacesOfTypeDictionariesCompletion)(NSArray *, NSString *type, NSError *);
 
 @implementation Place
 
-- (instancetype) initHubWithName: (NSString *)name
+#pragma mark - Initialization methods
+    
+- (instancetype) initHubWithName: (NSString *)name withArrayOfTypes:(NSArray *) arrayOfTypes
 {
     self = [super init];
-    NSArray *arrayOfTypes = [[NSArray alloc]initWithObjects:@"lodging", @"restaurant", @"museum", @"park", nil];
+    dispatch_async(dispatch_get_main_queue(), ^{
     dispatch_group_t serviceGroup = dispatch_group_create();
-    dispatch_group_enter(serviceGroup);
     
-    //CALL ONE - 1 CALL: GET HUB DICTIONARY FROM NAME
-    [[APIManager shared]getCompleteInfoOfLocationWithName:name withCompletion:^(NSDictionary *placeInfoDictionary, NSError *error) {
-        
-        // ** Call one result **
+    getNearbyPlacesOfTypeDictionariesCompletion getNearbyPlacesOfTypeDictionariesCompletionBlock = ^(NSArray *arrayOfPlacesDictionary, NSString *type, NSError *getPlacesError) {
+        if(arrayOfPlacesDictionary) {
+            NSArray* newArray = [arrayOfPlacesDictionary mapObjectsUsingBlock:^(id obj, NSUInteger idx) {
+                Place *place = [[Place alloc] initWithDictionary:obj];
+                if([self.dictionaryOfArrayOfPlaces objectForKey:type] == nil) {
+                    self.dictionaryOfArrayOfPlaces[type] = [[NSMutableArray alloc] init];
+                }
+                [self.dictionaryOfArrayOfPlaces[type] addObject:place];
+                return place;
+            }];
+        } else {
+            NSLog(@"ERROR IN GETTING DICTIONARIES OF NEARBY PLACES");
+        }
+        bool isDone = YES;
+        if((int)[self.dictionaryOfArrayOfPlaces count] != arrayOfTypes.count) {
+            isDone = NO;
+        }
+        if(isDone) {
+            dispatch_group_leave(serviceGroup);
+        }
```

Comment:
I'm a little confused why you need an isDone boolean here. Couldn't you just say:
if((int)[self.dictionaryOfArrayOfPlaces count] == arrayOfTypes.count) {
dispatch_group_leave(serviceGroup);
}

or if you need a boolean, you could do BOOL isDone = (int)[self.dictionaryOfArrayOfPlaces count] != arrayOfTypes.count;

^I guess my point is that the stuff inside the if statement is already a boolean value so you don't need to make another one.

Reply:
you have a point! it was actually something that I kept from a previous code, but I indeed do not need it anymore